pkg/cluster/cluster.go

```diff
@@ -75,17 +76,18 @@ func (c *Info) fetchAvailableAPIs(client *discovery.DiscoveryClient) error {
 }
 
 func (c *Info) fetchOpenShiftClusterVersion(ctx context.Context, cl client.Client) error {
-	key := client.ObjectKey{Name: "version"}
-	cversion := &configv1.ClusterVersion{}
-	if err := cl.Get(ctx, key, cversion); err != nil {
-		return fmt.Errorf("could not fetch ClusterVersion: %w", err)
+	cno := &configv1.ClusterOperator{}
+	err := cl.Get(ctx, types.NamespacedName{Name: "network"}, cno)
+	if err != nil {
+		return fmt.Errorf("error fetching OpenShift Cluster Network Operator: %w", err)
 	}
-	c.ID = string(cversion.Spec.ClusterID)
```

Comment:
We need either another way to get the c.ID, or keep on reading ClusterVersion as well. It's a way to automatically populate the cluster name or id, when multiCluster is enabled. If there's something else that can uniquely identify the cluster, it can be considered for a replacement.
The cluster ID can be overridden so it's not too bad if it doesn't work just with rosa, users will have workarounds